File: bubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;
// start of sorting function
template <typename T>
void bubbleSort(T& array,long long size,bool ascending=true) {
    for(long long i=0; i<size;i++){
        for(long long j=i+1; j<size;j++){
            // comparing elements
            if(array[i]>array[j]){
                if(ascending)
                swap(array[i],array[j]);
            }
            else{
                if(!ascending)
                swap(array[i],array[j]);
            }
        }
    }
}
// end of sorting function
int main(){
    long long array[]={1,2,3,4,5,6,7,3,2,2,4,4,5,32,2,23,4,423,4,8,4,0};
    bubbleSort(array,22);
    for(auto&it: array)cout<<it<<' ';
    cout<<endl;
    bubbleSort(array,22,false);
    for(auto&it: array)cout<<it<<' ';
}

File: bubbleSort.js
function bubbleSort(array, ascending = true) {
  for (let i = 0; i < array.length; i++) {
    for (let j = i + 1; j < array.length; j++) {
      if (array[i] > array[j]) {
        if (ascending) {
          let temp = array[i];
          array[i] = array[j];
          array[j] = temp;
        }
      } else {
        if (!ascending) {
          let temp = array[i];
          array[i] = array[j];
          array[j] = temp;
        }
      }
    }
  }
}
let array = [1,2,3,34,5,56,7,4,0,343,2,21,1,1,23,34,9,-3,986,7,53];
console.log(array);
bubbleSort(array);
console.log(array);
bubbleSort(array,false);
console.log(array);